torchx/schedulers/ray/ray_driver.py

```diff
@@ -127,17 +137,33 @@ def create_command_actors(
 
 
 def main() -> None:  # pragma: no cover
+
     actors: List[RayActor] = load_actor_json("actors.json")
+    if re.match("\\d+:\\d+", actors[0].nnodes_rep):
+        MIN_NNODES, MAX_NNODES = actors[0].nnodes_rep.split(":")
+        MIN_NNODES, MAX_NNODES = int(MIN_NNODES), int(MAX_NNODES)
+    else:
+        MIN_NNODES, MAX_NNODES = len(actors), len(actors)
     # pyre-fixme[16]: Module `worker` has no attribute `init`.
     ray.init(address="auto", namespace="torchx-ray")
-    pg: PlacementGroup = create_placement_group(actors)
-    command_actors: List[CommandActor] = create_command_actors(actors, pg)
+
+    created = 0  # number of placement group has been created
+    # Create the minimum requirement placement_group
+    pg: PlacementGroup = create_placement_group(actors[:MIN_NNODES])
+    command_actors: List[CommandActor] = create_command_actors(actors[:MIN_NNODES], pg)
+    created += MIN_NNODES
 
     active_workers = [
         command_actor.exec_module.remote()  # pyre-ignore
         for command_actor in command_actors
     ]
 
+    # keep creating placement groups until the maximum number of actors is reached
+    if created < MAX_NNODES:
```

Comment:
is this supposed to be a while loop or if?

we also don't increment created here though I'm wonder if it's better to delete this block and fold it into the below loop since we also have logic for it there